pykit/types.py

```diff
@@ -0,0 +1,157 @@
+        self.byValue = False
+        self.payload = {}
+    
+    def get_payload(self):
+        self.payload = self.parameterType.get_payload()
+        if self.isReturn:
+            self.payload['by'] = 'return'    
+        if self.byValue:
+            self.payload['by'] = 'val'    
+        
+        return self.payload
+    
+
+class ParameterType:
+    """
+    Object for IBM i ParameterTypes.
+    """
+    def __init__(self, name, value):
+        """
+        :param name:
+        :param value:
+        """
+        self.value = value
+        self.name = str(name)
+        self.payload = {"name":self.name}
+            
+            
+    def get_payload(self):
+        self.payload['value'] = self.value        
+        return self.payload
+    
+class Float(ParameterType):
+    """
+    Object for IBM i Float.
+    """
+    def __init__(self, name, length, precision, value):
+        """
+        How much error checking do we want here?
+        Should we check that length is positive?
+        Should we check if value matches length?
+        Should we truncate the value to the length?
+
+        :param length:
+        :param precision:s
+        :param value:
+        """
+        ParameterType.__init__(self, name, value)
+        self.length = length
+        self.precision = precision
+
+    def get_payload(self):
+        self.payload['type'] = str(self.length) + 'f' + str(self.precision)
+        ParameterType.get_payload(self)
+        return self.payload
+    
+
+class Integer(ParameterType):
+    """
+    Object for IBM i Integer.
+    """
+    def __init__(self, name, length, value, signed=True):
+        """
+        How much error checking do we want here?
+        Should we check that length is positive?
+        Should we check if value matches length?
+        Should we truncate the value to the length?
+
+        :param length:
+        :param value:
+        """
+        ParameterType.__init__(self, name, value)
+        self.length = length
+        self.signed = signed
+
+    def get_payload(self):
+        self.payload['type'] = str(self.length) + ('i' if self.signed else 'u') + '0'
+        ParameterType.get_payload(self)
+        
+        return self.payload
+
+
+class Character(ParameterType):
+    """
+    Object for IBM i Character.
+    """
+    def __init__(self, name, length, value, varying=''):
+        """
+        :param length:
+        :param value:
+        """
+        ParameterType.__init__(self, name, value)
+        self.length = length
+        self.varying = varying
+
+    def get_payload(self):
+        self.payload['type'] = str(self.length) + 'a'
+        if self.varying:
+            self.payload['type'] = self.payload['type'] + 'v' + str(self.varying)
+            
+        ParameterType.get_payload(self)
+        return self.payload
+
+
+class Decimal(ParameterType):
+    """
+    Object for IBM i Decimal.
+    """
+    def __init__(self, name, length, decimals, value, signed=False):
```

Comment:
Huh, apparently indeed the `s` does stand for "signed". There's also an "l" and "r" types which have a leading or trailing sign digit separate from the numeric data (I wonder if anyone actually uses those?). I don't know of anybody that refers to them as "signed decimal" though (could be showing ignorance here) - I've always heard it referred to as zoned decimal. Indeed the the free-format names use ZONED and PACKED. Perhaps instead, we just flip the condition and use `packed=True`?

Alternatively instead of a parameter, use separate classes: `ZonedDecimal` and `PackedDecimal` (or `Numeric` and `Decimal`) You could use this class as an internal base class.